### DISCUSSION TITLE
Améliorer/Revoir la distribution des points du bonus de ligue

### DISCUSSION THREAD

**Discussion by Ntalcme** [Crownicles/Crownicles#2519] [Nouvelles idées]
Après quelques petits tests avec la formule qui gère la distribution des points en fonction du rang, je me rend compte qu'elle n'est pas très juste envers tous les joueurs. Ici je ne remets pas en cause la quantité distribuée mais l'écart entre les distributions de points.
## Explication
Je vais prendre quelques rangs en exemple : 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 et 20.
Au rang 1, nous gagnons 4 000 🏅 
Au rang 2 : 3 190 🏅 
Rang 3 : 2 950 🏅 
Rang 4 : 2 830 🏅 
Rang 5 : 2 760 🏅 
Rang 6 : 2 700 🏅 
Rang 7 : 2 660 🏅 
Rang 8 : 2 630 🏅 
Rang 9 : 2 600 🏅 
Rang 10 : 2 570 🏅 
Rang 20 : 2 380 🏅 

(en accord avec la formule [ici](https://github.com/DraftBot-A-Discord-Adventure/DraftBot/blob/master/src/core/database/game/models/Player.ts#L924) ligne 924 à 937)

### Maintenant, regardons les écarts.
Écart entre le rang 1 et rang 2 : 810 🏅(rang d'écart : 1)
Écart entre le rang 2 et rang 3 : 240 🏅 (rang d'écart : 1)
Écart entre le rang 3 et rang 4 : 120 🏅 (rang d'écart : 1)
Écart entre le rang 4 et rang 5 : 70 🏅 (rang d'écart : 1)
Dans le même schéma, on a en écart dans l'ordre jusqu'au rang 10 : 60, 40, 30, 30, 30.

On constate à peu près des écarts répétitifs entre le rang 11 et 20 également (variant entre 10 et 20).
(pareil entre le rang 21 et 30 mais je ne suis pas allé plus loin non plus)

Intéressons nous à un dernier écart, le rang 2 et 20 : 810 🏅 
C'est exactement le même écart qu'entre le top 1 et 2 mais avec ici **18** rang de différence.

**La question qu'on peut se poser ici est la suivante, le top 1 gagne-t-il trop ou bien les rangs suivants ne gagnent pas assez ?**

### Conclusion
Si l'on se base sur les écarts qu'on peut trouver rang à rang plus bas, le top 1 gagne trop.
Si l'on se base sur l'implication des joueurs, les rangs proches du top 1 ne gagnent pas assez.

Dans tous les cas, un truc cloche et je pense qu'il faudrait améliorer tout ça.

⚠️ NE ME RAMENEZ PAS LE TW DANS L'HISTOIRE, LÀ N'EST POINT LE SUJET.

**Comment by aureochocob0n** [comment]
Il faudrait donc trouver une autre formule?

**Comment by aureochocob0n** [comment]
Formule de gpt: 
![Screenshot_20240621_161737_Chrome](https://github.com/DraftBot-A-Discord-Adventure/DraftBot/assets/153011168/259c772e-70a3-48bf-8da9-3936bff39d9b)
![Screenshot_20240621_161715_Chrome](https://github.com/DraftBot-A-Discord-Adventure/DraftBot/assets/153011168/8a417a14-fc74-4dff-a2ed-09f1e1f21a66)
